Fix KeyboardInterrupt handling in QuasiXML parser

stsnel opened the pull request:
Catch and raise KeyboardInterrupts in the QuasiXML parser, so
that the parser does not throw a parser exception if the user
presses ctrl-C when XML data is being processed.

cjsmeele commented:
I agree that this is an issue, though I do not agree with the solution -- protocol handling code within a library should not have any knowledge of a keyboard / tty being used in the main program.
Instead, the caught exception should be made more specific into `Exception` (instead of implicit `BaseException`, because `BaseException` is used for other events external to the code being tried).

stsnel committed:
Fix exception handling QuasiXML parser

Catch Exceptions rather than implicitly catching BaseExceptions,
so that KeyboardInterrupts don't result in a QuasiXmlParseError.

stsnel commented:
> I agree that this is an issue, though I do not agree with the solution -- protocol handling code within a library should not have any knowledge of a keyboard / tty being used in the main program.
> Instead, the caught exception should be made more specific into `Exception` (instead of implicit `BaseException`, because `BaseException` is used for other events external to the code being tried).

Thanks for your feedback. I've changed the PR.
